<launch>
    <arg name="bag_filename"/>
    <arg name="dst_directory" default="/tmp/odometry"/>
    <arg name="points_topic" default="/velodyne_points"/>
    <arg name="odom_topic" default="/odom"/>
    <arg name="endpoint_frame" default="velodyne"/>
    <arg name="origin_frame" default="odom"/>

    <param name="use_sim_time" value="true"/>

    <node pkg="odometry_saver" type="odometry_saver" name="odometry_saver" output="screen">
        <param name="dst_directory" value="$(arg dst_directory)"/>
        <param name="origin_frame" value="$(arg origin_frame)"/>
        <param name="endpoint_frame" value="$(arg endpoint_frame)"/>
        <remap from="/points" to="$(arg points_topic)"/>
        <remap from="/odom" to="$(arg odom_topic)"/>
    </node>

    <include file="$(find hdl_graph_slam)/launch/hdl_graph_slam_400.launch"/>
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find hdl_graph_slam)/rviz/hdl_graph_slam.rviz"/>

    <!-- bag_player -->
    <node pkg="hdl_graph_slam" type="bag_player.py" name="bag_player" args="$(arg bag_filename)" output="screen"/>
</launch>
